src/test/java/com/brandwatch/internship/fetchandstoretweets/integrationtests/QueryIT.java

```diff
@@ -0,0 +1,108 @@
+package com.brandwatch.internship.fetchandstoretweets.integrationtests;
+
+import com.brandwatch.internship.fetchandstoretweets.entities.Query;
+import com.brandwatch.internship.fetchandstoretweets.integrationtests.utility.UrlCreator;
+import com.brandwatch.internship.fetchandstoretweets.repositories.QueryRepository;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.io.IOException;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@TestPropertySource(locations="classpath:test.properties")
+public class QueryIT {
+
+    private static final long QUERY_ID = 3;
+    private static final String BASE_URI = "/queries";
+
+    @LocalServerPort
+    private int port;
+
+    @Autowired
+    private QueryRepository queryRepository;
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    private UrlCreator urlCreator;
+    private TestRestTemplate restTemplate = new TestRestTemplate();
+
+    @Before
+    public void setUp() {
+        urlCreator = new UrlCreator(restTemplate, BASE_URI, port);
+        restTemplate.getRestTemplate().setRequestFactory(
+                new HttpComponentsClientHttpRequestFactory(HttpClientBuilder.create().build()));
+    }
+
+    @Test
+    public void getAllQueriesTest() throws IOException {
+        final String uri = "/";
```

Comment:
Maybe you could make `uri` an instance variable, rather than a local variable as you reference it a lot?

Reply:
I can add it to the `BASE_URI`  variable but I think that URIs and paths should not end with slashes and whenever we want to append to the URI, we should add this slash at the beginning of the new string. `uri` by the way is not the same in all of the test methods as I append to the `"/"` the `QUERY_ID` in two of them.